### PR TITLE
:sparkles: Refreshed whole project

### PR DESCRIPTION
**Pull request by ujas-m-simformsolutions**
Update template with below points
*  Fix xcode_backend.sh not found issue
* Add proper Splash screen (add simform logo)
* Network Stream example
* Add Error handling to Repository
* Added flutter_intl for localization
* Add asset gen (flutter_gen)
* Fix IOS orientation not applied natively
* Replace network check function with stream or add separately
* Create all singleton properly for constants and strings
* Create enum for shared prefs keys
* Update SharedPrefs flow
* Create function to create route and pass route settings in it
* Add routes generator to onGenerateRoutes
* Remove unnecessary add provider for helper (already in extensions)
* Add secure storage for tokens
* Remove `^` symbol from packages in pubspec.yaml
* Update flutter version and package version
* Replace AppStrings with Constants and add approutes
* Add navigator key
* Remove getIt
* Rename AppConstants -> Constants
* Add flutterTarget in flavor.gradle

### CODE REVIEW

File: {{cookiecutter.repo_name}}/lib/modules/splashScreen/splash_screen.dart [hunk 1]
```diff
@@ -2,22 +2,22 @@ import 'dart:async';
...
-  const SplashScreen({Key? key}) : super(key: key);
+  const SplashScreen({super.key});
 
   @override
   _SplashScreenState createState() => _SplashScreenState();
 }
 
 class _SplashScreenState extends State<SplashScreen> {
-  final splashDelay = 5;
+  final splashDelay = 3;
```
Comment (dhvanit-v-simformsolutions) [inline {{cookiecutter.repo_name}}/lib/modules/splashScreen/splash_screen.dart:20]: If we use this variable at one place we should assign directly and if required we can add comment above the line what is the use case.

File: {{cookiecutter.repo_name}}/lib/routes.dart [hunk 2]
```diff
@@ -1,22 +1,34 @@
...
+
 import 'modules/splashScreen/splash_screen.dart';
 import 'modules/splashScreen/splash_screen_store.dart';
+import 'utils/common_widgets/invalid_route.dart';
 import 'utils/extensions.dart';
+import 'values/strings.dart';
 
 class Routes {
   static Route<dynamic> generateRoute(RouteSettings settings) {
+    Route<dynamic> getRoute({
```
Comment (dhvanit-v-simformsolutions) [inline {{cookiecutter.repo_name}}/lib/routes.dart:11]: Make this function private if this is not used outside of scope

File: {{cookiecutter.repo_name}}/lib/services/secure_storage_service.dart [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import 'package:flutter_secure_storage/flutter_secure_storage.dart';
```
Comment (dhvanit-v-simformsolutions) [inline {{cookiecutter.repo_name}}/lib/services/secure_storage_service.dart:1]: I guess same file is exists with different name "secure_storage_helper.dart"

Comment (dhvanit-v-simformsolutions) [reply {{cookiecutter.repo_name}}/lib/services/secure_storage_service.dart:1]: Remove any of one file

File: {{cookiecutter.repo_name}}/lib/utils/extensions.dart [hunk 4]
```diff
@@ -60,7 +60,7 @@ extension StatefulWidgetExtension on State {
 
 /// allows to create [MultiProvider] with less boilerplate
 extension ProviderExtension<T> on Widget {
-  Widget withProviders(List<Provider> providers) => MultiProvider(
+  Widget withProviders(List<Provider<T>> providers) => MultiProvider(
```
Comment (dhvanit-v-simformsolutions) [inline {{cookiecutter.repo_name}}/lib/utils/extensions.dart:63]: Remove sdp extension or you can refer to regline project as we are not use this anymore. Also add few other extension like isNotNullNotEmpty. 
You can refer regline project for same

File: {{cookiecutter.repo_name}}/lib/apibase/header_interceptor.dart [hunk 5]
```diff
@@ -2,8 +2,9 @@ import 'dart:async';
 
 import 'package:dio/dio.dart';
 
+import '../services/secure_storage_service.dart';
 import '../utils/network_utils.dart';
-import '../utils/shared_preferences_helper.dart';
+import '../values/enumeration.dart';
```
Comment (dhvanit-v-simformsolutions) [inline {{cookiecutter.repo_name}}/lib/apibase/header_interceptor.dart:7]: I guess we should use package imports instead relative one. Please ask to any other developer for same and change accordingly.

**Review by dhvanit-v-simformsolutions — CHANGES_REQUESTED** [review]
Please make chages

File: {{cookiecutter.repo_name}}/lib/routes.dart (hunk 2 above)
Comment (ujas-m-simformsolutions) [reply {{cookiecutter.repo_name}}/lib/routes.dart:11]: As this is inside another function linter suggests to not use `_` for this as it scope is already limited.

File: {{cookiecutter.repo_name}}/lib/apibase/header_interceptor.dart (hunk 5 above)
Comment (ujas-m-simformsolutions) [reply {{cookiecutter.repo_name}}/lib/apibase/header_interceptor.dart:7]: The relative imports are recommended by dart team. This is also added in the flutter bible and in official dart docs [here](https://dart.dev/tools/linter-rules#prefer_relative_imports).

File: {{cookiecutter.repo_name}}/lib/services/shared_preferences_service.dart [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+import 'package:shared_preferences/shared_preferences.dart';
+import '../values/enumeration.dart';
+
+class SharedPreferencesService {
+  SharedPreferencesService._();
+
+  static final SharedPreferencesService _instance =
+  SharedPreferencesService._();
+
+  static SharedPreferencesService get instance => _instance;
```
Comment (dhvanit-v-simformsolutions) [inline {{cookiecutter.repo_name}}/lib/services/shared_preferences_service.dart:11]: Can we directly assign this SharedPreferencesService._() here ?

File: {{cookiecutter.repo_name}}/lib/services/shared_preferences_service.dart [hunk 7]
```diff
@@ -0,0 +1,53 @@
+import 'dart:async';
+import 'package:shared_preferences/shared_preferences.dart';
+import '../values/enumeration.dart';
+
+class SharedPreferencesService {
+  SharedPreferencesService._();
+
+  static final SharedPreferencesService _instance =
```
Comment (dhvanit-v-simformsolutions) [inline {{cookiecutter.repo_name}}/lib/services/shared_preferences_service.dart:8]: Referring below comment can we remove this line ?

File: {{cookiecutter.repo_name}}/lib/services/secure_storage_service.dart [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+///This is a helper class to store sensitive information, for example auth
+///tokens.
+class SecureStorageService {
+  SecureStorageService._();
+
+  final FlutterSecureStorage _storage = const FlutterSecureStorage();
+
+  static final SecureStorageService _instance = SecureStorageService._();
+
+  static SecureStorageService get instance => _instance;
```
Comment (dhvanit-v-simformsolutions) [inline {{cookiecutter.repo_name}}/lib/services/secure_storage_service.dart:13]: Can we directly assign this SecureStorageService._() here ?

File: {{cookiecutter.repo_name}}/lib/services/secure_storage_service.dart [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+import '../values/enumeration.dart';
+
+///This is a helper class to store sensitive information, for example auth
+///tokens.
+class SecureStorageService {
+  SecureStorageService._();
+
+  final FlutterSecureStorage _storage = const FlutterSecureStorage();
+
+  static final SecureStorageService _instance = SecureStorageService._();
```
Comment (dhvanit-v-simformsolutions) [inline {{cookiecutter.repo_name}}/lib/services/secure_storage_service.dart:11]: Referring below comment can we remove this line ?

File: {{cookiecutter.repo_name}}/lib/utils/network_utils.dart [hunk 10]
```diff
@@ -1,24 +1,23 @@
...
 
-  dynamic checkInternet(Function(bool) func) {
-    checkIsInternet().then((internet) {
-      if (internet) {
-        func(true);
-      } else {
-        func(false);
-      }
-    });
+  static final NetworkUtils _instance = NetworkUtils._initialise();
```
Comment (dhvanit-v-simformsolutions) [inline {{cookiecutter.repo_name}}/lib/utils/network_utils.dart:13]: ^^

**Review by dhvanit-v-simformsolutions — CHANGES_REQUESTED** [review]
Please make changes

**Review by dhvanit-v-simformsolutions — APPROVED** [review]

File: {{cookiecutter.repo_name}}/lib/app.dart [hunk 11]
```diff
@@ -1,28 +1,35 @@
...
-        primaryColor: AppColors.colorPrimary,
-        colorScheme:
-            ColorScheme.fromSwatch().copyWith(secondary: AppColors.colorAccent),
-        unselectedWidgetColor: AppColors.colorPrimary,
-      ),
-      home: const SplashScreen(),
-      onGenerateRoute: Routes.generateRoute,
-      navigatorKey: locator<NavigationService>().navigatorKey,
+    final themeStore = context.provide<AppThemeStore>();
+    return Observer(
```
Comment (vatsaltanna-simformsolutions) [inline {{cookiecutter.repo_name}}/lib/app.dart:21]: please make these changes based on the cookie-cutter flag. add a new flag for theme support

File: {{cookiecutter.repo_name}}/lib/app.dart [hunk 12]
```diff
@@ -1,28 +1,35 @@
...
 
-/// Change name of app as per Project
-class WellnessApp extends StatelessWidget {
-  const WellnessApp({Key? key}) : super(key: key);
+final GlobalKey<NavigatorState> navigatorKey = GlobalKey();
+
+NavigatorState get navigator => navigatorKey.currentState!;
+
+// TODO(user): Change name of app as per Project
+class SimformApp extends StatelessWidget {
```
Comment (vatsaltanna-simformsolutions) [inline {{cookiecutter.repo_name}}/lib/app.dart:15]: change this to the APP name

File: {{cookiecutter.repo_name}}/lib/app_config.dart [hunk 13]
```diff
@@ -1,14 +1,16 @@
 import 'package:flutter/material.dart';
 import 'package:flutter/services.dart';
+import 'package:provider/provider.dart';
 
 import 'app.dart';
 import 'utils/extensions.dart';
+import 'values/app_theme_store.dart';
 
 /// represents the pixel ratio of the device
 double devicePixelRatio = 0;
```
Comment (vatsaltanna-simformsolutions) [inline {{cookiecutter.repo_name}}/lib/app_config.dart:10]: remove this as it is unnecessary.

File: {{cookiecutter.repo_name}}/lib/app_config.dart [hunk 14]
```diff
@@ -19,7 +21,10 @@ class AppConfig extends StatelessWidget {
     return GestureDetector(
       onTap: context.hideKeyboard,
       child: Builder(
-        builder: (ctx) => const WellnessApp(),
+        builder: (ctx) => Provider(
+          create: (_) => AppThemeStore(),
+          child: const SimformApp(),
```
Comment (vatsaltanna-simformsolutions) [inline {{cookiecutter.repo_name}}/lib/app_config.dart:26]: use extension.

File: {{cookiecutter.repo_name}}/lib/flavors/flavor_config.dart [hunk 15]
```diff
@@ -5,11 +5,11 @@ import '../flavors/flavor_values.dart';
 
 class FlavorConfig extends InheritedWidget {
   FlavorConfig({
-    Key? key,
     required this.flavor,
     required this.values,
+    super.key,
     Widget? child,
-  }) : super(key: key, child: child!);
+  }) : super(child: child!);
```
Comment (vatsaltanna-simformsolutions) [inline {{cookiecutter.repo_name}}/lib/flavors/flavor_config.dart:12]: use super.child

File: {{cookiecutter.repo_name}}/lib/main_prod.dart [hunk 16]
```diff
@@ -26,16 +26,16 @@ void main() {
...
-  const FlavoredApp({Key? key}) : super(key: key);
+  const FlavoredApp({super.key});
 
   @override
   Widget build(BuildContext context) {
     return FlavorConfig(
       flavor: Flavor.prod,
-      values: FlavorValues(baseUrl: '{{cookiecutter.base_url}}'),
+      values: FlavorValues(
+        baseUrl: '{{cookiecutter.base_url}}',
```
Comment (vatsaltanna-simformsolutions) [inline {{cookiecutter.repo_name}}/lib/main_prod.dart:36]: appconstant

File: {{cookiecutter.repo_name}}/lib/modules/splashScreen/splash_screen.dart [hunk 17]
```diff
@@ -27,60 +25,75 @@ class _SplashScreenState extends State<SplashScreen> {
...
-                  .of(context)
-                  .size
-                  .width,
-            ),
-          ],
-        ),
+    return ColoredBox(
+      color: Colors.white,
+      child: Assets.images.splash.image(
+        height: MediaQuery.of(context).size.height,
```
Comment (vatsaltanna-simformsolutions) [inline {{cookiecutter.repo_name}}/lib/modules/splashScreen/splash_screen.dart:40]: use extension

File: {{cookiecutter.repo_name}}/lib/modules/splashScreen/splash_screen.dart [hunk 18]
```diff
@@ -27,60 +25,75 @@ class _SplashScreenState extends State<SplashScreen> {
...
+        width: MediaQuery.of(context).size.width,
       ),
     );
   }
 }
 
-class AfterSplash extends StatelessWidget {
-  const AfterSplash({Key? key}) : super(key: key);
+class AfterSplash extends StatefulWidget {
```
Comment (vatsaltanna-simformsolutions) [inline {{cookiecutter.repo_name}}/lib/modules/splashScreen/splash_screen.dart:47]: please check this

**Pull request merged**